billing/plan/service.go

```diff
@@ -327,3 +338,23 @@ func verifyDuplicatePlans(planFile File) error {
 	}
 	return nil
 }
+
+func initializeProductFeatureMap(p []Plan) map[string][]product.Feature {
+	productFeatures := map[string][]product.Feature{}
+	for _, pln := range p {
+		products := pln.Products
+		for _, prod := range products {
+			productFeatures[prod.ID] = []product.Feature{}
+		}
+	}
+	return productFeatures
+}
+
+func mapFeaturesToProducts(productFeatureMap map[string][]product.Feature, features []product.Feature) {
```

Comment:
nit: IMO creating functions for 3-4 lines being used only at one place is not much of a use and code is easier to read when it's a single consistent flow. Also it's better to not modify the args, and instead return the new value.

Reply:
> Also it's better to not modify the args, and instead return the new value.
Agreed, updated to not do that

In terms of methods, I believe smaller methods which perform non-functional conversions/translations can sometimes distract from the actual logic of what the method is doing. Hence, separated out the mapping into a separate method. It can also help when debugging since there is a clear separation of parts in the method.